### ISSUE TITLE
Bug: Containers with network dependencies fail to start from time to time 

### ISSUE THREAD

**Issue by AN1MATEK** [Commifreak/unraid-appdata.backup#38]
Hello there,

This issue happens only sometimes, like once every 10/15 backups.

I have a container that is a VPN tunnel, PIA. There are 3 other containers that have `Network Type: Custom container:pia`

The VPN container starts always first. When the other 3 start after the backup, they fail, as they are looking for a container id for their network and this has changed. I have a stupid fix in place in the post-run script that checks if the containers are running or not, and then executes the whole `docker run`, as it is displayed when starting from the UI. They come up without an issue. 

```
if docker ps --filter "name=thelounge" --filter "status=running" | grep -q "thelounge"; then
    echo -e "Container thelounge is running\n"
else
    echo -e "Container thelounge is not running.\n"
    /usr/bin/docker remove thelounge
    /usr/bin/docker run -d --name='thelounge' --net='container:pia' [...]'
```

Since I have this fix in my script I don't have issues anymore, other than the red notifications when it happens. I wanted to bring attention to it.

Thank you for the development efforts. The plugin is staple for me.

Debug log ID: `4b60e7f5-421c-45b4-abcf-74730e670f60`

**Comment by berubejp** [comment]
I have the same issue. I've got containers behind a vpn (with --net='container:myvpncontainer') and they all fail to restart after backup most of the time (maybe 75%). 
I also created a dumb script that would `docker start xx` them after the backup but I get a `No such container` error. Strangely, if I browse to the Unraid docker page then start the script, it works...

**Comment by AN1MATEK** [comment]
As you see in my script, you have to remove the old container and do `docker run ...`. Just copy what I did and it will work.

**Comment by berubejp** [comment]
Yeah that's what I did and it worked, thanks. But having to keep configurations updated in 2 places is not perfect.

**Comment by PilaScat** [comment]
same issue
debug id 868ba415-051b-42e2-b406-3f626a882054

**Comment by b3rrytech** [comment]
I also have this exact same problem. Trying to find a sustainable solution to it and looking at options.

One way of getting a reliable solution for this, other than running AN1MATEK’s script, would probably be to convert these containers to one Docker Compose stack and use the [“depends_on”](https://docs.docker.com/compose/how-tos/startup-order/) attribute.

One could use the [Comporize](https://github.com/composerize/composerize) tool to convert the Docker run command seen during installation/edit and then create a stack with both the VPN container and the dependent container(s). Since it converts the original CA Docker run it should result in a very similar Docker tab with correct icons etc.

How will this plugin handle this “depends_on” attribute if the backup type is set to “stop, backup and start each container” @Commifreak?

**Comment by AN1MATEK** [comment]
Hey @b3rrytech looking more into this the best way possible imo is to parse the container's xml file into a `docker run` command. So you would take the file in `/boot/config/plugins/dockerMan/templates-user/my-container.xml` and somehow parse it and insert it in you command. That way it is automated and you don't need to maintain 2 places.

If somebody can create a function to do that, it might be PR material - or at least helps folks that come here.
